src/rust/engine/process_execution/src/cache.rs

```diff
@@ -106,23 +111,20 @@ impl CommandRunner {
     use bazel_protos::remote_execution::ExecuteResponse;
     let file_store = self.file_store.clone();
 
-    let platform_map = self.platform_map.clone();
-
     self
       .process_execution_store
       .load_bytes_with(fingerprint.clone(), move |bytes| {
+        let decoded: PlatformAndResponseBytes = bincode::deserialize(&bytes[..])
+          .map_err(|err| format!("Could not deserialize platform and response: {}", err))?;
```

Comment:
The breaking schema change thing is interesting... I _think_ the only thing people are actually using v2 process execution for in the wild is cloc, and it doesn't feel like super the end of the world for them to not be able to get cache hits... It's a shame that this would introduce errors for them, rather than just ignore the bad entries, but that's probably better than ignoring bad entries looking forwards...

It may be worth making this logging about possible corruption and returning `None` (so that the process will be re-run) rather than an error?

Reply:
Something else we could do is change the location on disk of the LMDB we use, or use a different "database" within the same location, so that we're just not clashing.

Reply:
Negative on it only being cloc. We know users are using V2 linters like Black. 

Greg was discussing yesterday some different ways to approach a version change and I think may use one of those approaches here. (I didn’t closely follow the conversation)

Reply:
I'm going to hold off on merging this until https://github.com/pantsbuild/pants/pull/9269 is reviewed and merged